src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -100,6 +101,75 @@ def pileupSizeTask(self):
             self.logger.exception(msg)
         return cmsDict, cusDict
 
+    def partialPileupTask(self, rseExpression='T1_FNAL_Disk|T2_CERN_Disk'):
+        """
+        Execute partial pileup placement according to the following logic:
+
+        - get rucio DIDs (datasets, in CMS name blocks) for our pileup document
+        - get portion of DIDs based on ceil(containerFraction * num_rucio_datasets)
+        - create new container DID in Rucio via Rucio wrapper createContainer API
+          which calls rucio CLI add_container
+        - update MSPileup JSON with
+          - create new custom Name as pileup+extention
+          - add new transition record and update MSPileup document
+        - we call attachDIDs Rucio wrapper API with our set of DIDs
+          - check if rse is mandatory in attachDIDs API, we prefer to not pass it, otherwise
+          we will pass rses from pileup document
+          - create new rules for custom DID using either FNAL disk or T2_CERN sites
+          - set expiration date (to be 24h) for already existing ruleIds from pileup document
+          - add new ruleIds to pileup document
+
+        :param rseExpression: string, Rucio rse expression, see
+        https://rucio.cern.ch/documentation/started/concepts/rse_expressions/
+        :return: None
+        """
+        spec = {'containerFraction': {"$ge": 0}}
```

Comment:
`gt`? There shouldn't be any containerFraction=0.0

Reply:
done